### PR TITLE
Bugfix note text without formatting tags

### LINKED ISSUE #84: Note output as text gives the entire xml element
The method  **public String findTextById(Long id) {...}** from the class **NoteDao.java** gives as output the entire xml element and not the note's text value. Is it supposed to give the entire xml element or only the text of the note?

Here is my console output:
`2021-02-25 16:36:34.982 25571-25571/de.bibbuddy I/System.out: NoteText     <p dir="ltr" style="margin-top:0; margin-bottom:0;">my text note</p>`

Please take a look into this method.

### PR DESCRIPTION
**Pull request by LuisMossburger**
**Description**
Closes #84.
Adds a new function which strips formatting tags from note text. This is used for the export function from now on.

**Affects**
Export function and Note.

**Notes for Reviewer**
Please review & comment/approve. Thanks!
See this example to review the new regex: https://regex101.com/r/V3CgIs/1

### CODE REVIEW

**Comment by LuisMossburger** [comment]
@silvia-ivanova-github-rep Please wait for @xSabrina's next PR to be merged before reviewing so no merge conflicts arise for the pretty big next PR from her :) Thanks!

File: BibBuddy/app/src/main/java/de/bibbuddy/NoteDao.java
Comment (silvia-ivanova-github-rep) [inline BibBuddy/app/src/main/java/de/bibbuddy/NoteDao.java:250]: You should not replace all characters, because if the user uses some of those characters in the note, then this note's text value is not shown properly in the exported BibTeX-file. Maybe you should set limits for replaced chars.

**Review by silvia-ivanova-github-rep — CHANGES_REQUESTED** [review]

File: BibBuddy/app/src/main/java/de/bibbuddy/NoteDao.java
Comment (LuisMossburger) [reply BibBuddy/app/src/main/java/de/bibbuddy/NoteDao.java:250]: Hey, thanks for the suggestion! You're completely right here, however, your proposed solution won't work as this would strip "short user typed tags" as well. I will therefore see if I can specify the exact output of the rich editor to only strip those tags.

**Review by silvia-ivanova-github-rep — CHANGES_REQUESTED** [review]
Thanks for the changes. Unfortunately, I still have the same issue as before this commit. I created a note with text value of "3<4"  and then I exported the .bib file. The text value of the note is written in the .bib-File as "3&It;4". Also after exporting the file with this note and then importing it again, the note is named with the name of the shelf and not with the content of the note.  After updating the Fragment where the notes are listed, the correct names of the notes are shown. Could you please test it again? If you cannot find where the error is, I can take a look in the next few days.

**Comment by LuisMossburger** [comment]
Hey, thanks for the review! As far as I can tell, both problems do not arise from the code I wrote, however.

1) "&lt;" instead of ">" arises because special signs are encoded when stored into the database. I can decode that for the export, but that should be discussed in the team meeting I think.

2) Import of note - that is a separate bug probably somewhere in the "import" section you wrote yourself? I did not change anything regarding import functions.

**Review by silvia-ivanova-github-rep — APPROVED** [review]

**Comment by LuisMossburger** [comment]
For the protocol: We had a chat and think we should create new issues for both of these problems, as they are not directly related to the original bug here. This PR will be merged after #117 is.

**Comment by LuisMossburger** [comment]
Pr #117 will be closed, so this can be merged!

**Pull request merged**
